packages/authentication/test/unit/provider.test.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, Provider, ValueOrPromise, instantiateClass} from '@loopback/context';
+import {HttpErrors, ParsedRequest} from '@loopback/core';
+import {AuthenticationProvider, getAuthenticatedUser, AuthenticateFn, UserProfile} from '../..';
+import {MockStrategy} from './fixtures/mock-strategy';
+
+describe('AuthenticationProvider', () => {
+  let provider: Provider<AuthenticateFn>;
+  let strategy: MockStrategy;
+
+  const mockUser: UserProfile = {name: 'mock-user', locale: 'mock-locale'};
+
+  beforeEach(givenAuthenticationProvider);
+
+  describe('constructor()', () => {
+    it('instantiateClass injects authentication.strategy in the constructor', async () => {
+      const context: Context = new Context();
+      context.bind('authentication.strategy').to(strategy);
+      provider = await instantiateClass<Provider<AuthenticateFn>>(AuthenticationProvider, context);
```

Comment:
I think these three lines (22-24) are duplicating the work already done by `givenAuthenticationProvider`, aren't it? Can you remove it please?